tests/model/test_model.py

```diff
@@ -1528,6 +1528,57 @@ def test_fetch_raw_message_content(
         assert self.display_error_if_present.called == display_error_called
         assert return_value == expected_raw_content
 
+    @pytest.mark.parametrize(
+        "response, expected_read_receipts, feature_level",
+        [
+            case(
+                {"msg": "", "result": "success", "user_ids": [3, 7, 9]},
+                [3, 7, 9],
+                139,
+                id="valid_ZFL_with_valid_response",
+            ),
+            case(
+                {"msg": "Invalid message(s)", "result": "error"},
+                None,
+                137,
+                id="valid_ZFL_with_invalid_response",
+            ),
+        ],
+    )
+    def test_fetch_message_read_receipt_user_ids_valid_zfl(
+        self,
+        mocker,
+        model,
+        expected_read_receipts,
+        response,
+        feature_level,
+        message_id=1,
+    ):
+        model.server_feature_level = feature_level
+        self.client.call_endpoint.return_value = response
+        result = model.fetch_message_read_receipt_user_ids(message_id)
+        assert result == expected_read_receipts
+
+    @pytest.mark.parametrize(
+        "feature_level",
+        [
+            case(
+                136,
+                id="invalid_ZFL_with_response",
+            ),
```

Comment:
If there is only one case, then I would normally suggest it be inline instead, often with parameters such as eg. the feature level able to be specified as a default parameter to the test function - and the id is the 'extra' name of the test.

However, I'd suggest you could also cover back to the minimum feature level, which is typically `None` (and will shortly be `0` after a refactor in another PR).

Reply:
Adding a `None` case in the invalid zfl test function.

Also I understand keeping it inline for just one case saves space but it just looks so much cleaner and more consistent this way with id and all, right?

Reply:
I paused reviewing this PR since I wanted to get #1462 merged first, which will make these values zero instead of `None`.